File: pgm/ABC_Java/ABC_259_A.java
import java.io.*;
// import java.math.*;
// import java.util.*;
// import java.util.Map.*;
// import java.util.stream.*;

public class Main {
    public static void main (String[] args) throws Exception {
        BufferedReader reader =
            new BufferedReader (new InputStreamReader (System.in));
        final String[] arr = reader.readLine().split (" ");
        final int M = Integer.parseInt (arr[1]);
        final int X = Integer.parseInt (arr[2]);
        final int T = Integer.parseInt (arr[3]);
        final int D = Integer.parseInt (arr[4]);
        System.out.println (M <= X ? T - D * (X - M) : T);
        return;
    }
}

File: pgm/ABC_Kotlin/ABC_259_A.kt
// import java.io.BufferedOutputStream
// import java.io.BufferedReader
// import java.io.DataInputStream
// import java.io.FileInputStream
// import java.io.IOException
// import java.io.InputStreamReader
// import java.io.PrintWriter
// import java.lang.StringBuilder
// import java.time.*
// import java.util.Scanner
// import java.util.StringTokenizer

import kotlin.math.*

const val space: String = " "
const val zero: Int = 0

fun main() {
    val (N: Int, M: Int, X: Int, T: Int, D: Int) = readln().split(space).map { it.toInt() }
    println(T + min(zero, (M - X) * D))
    kotlin.system.exitProcess(0)
}